### PR TITLE
Solved the mermaid diagram of #36

### PR DESCRIPTION
**Pull request by Danuragtiwari**
solved the mermaid diagram, Attaching ss below.!
<be>
Screenshot (1)
<be>
Screenshot (2)
<be>
Screenshot (3)
<be>
Screenshot (4)
<be>
let me know if any changes needed :)

### CODE REVIEW

**Comment by Danuragtiwari** [comment]
@harshil21 please once have a look!

File: README.md [hunk 1]
```diff
@@ -57,10 +57,87 @@ AirbrakesV2/
 │   ├── ...
 ├── main.py [main file used to run on the rocket]
 ├── constants.py [file for constants used in the project]
-├── pyproject.toml [configuration file for the project]
+├── pyproject.toml [confipip install mkdocs-mermaid2-pluginguration file for the project]
```
Comment (harshil21) [inline README.md:60]:
```suggestion
├── pyproject.toml [configuration file for the project]
```

revert this change. The mermaid diagram is only meant to be seen on github so it's okay not having that dependency

File: README.md [hunk 2]
```diff
@@ -57,10 +57,87 @@ AirbrakesV2/
...
+      mainpy((main.py)):::circular --> Airbrakes((Airbrakes Context)):::circular
+     
+    end
+    Airbrakes --> Update[update]:::bubble
+    %% IMU Data Packet Flow
+    IMUDataPacket --> Update
+    
+    %% States as individual nodes
+    State((State)):::circular
+    Flightstate((Flight)):::circular
```
Comment (harshil21) [inline README.md:87]:
```suggestion
    CoastState((Flight)):::circular
```
sorry I didn't write the correct state on my diagram. It's called as CoastState now instead of FlightState. Make sure to update the other references too.

File: README.md [hunk 3]
```diff
@@ -57,10 +57,87 @@ AirbrakesV2/
...
+    Update --> LogData[(Logged Data Packet)]:::outputSquare
+    LogData --> LogFunction
+
+    %% Airbrake Control Methods with Parentheses Displayed
+    Update --> ExtendAirbrakes[extend_airbrakes]:::bubble
+    Update --> RetractAirbrakes[retract_airbrakes]:::bubble
+
+    %% Servo Connections
+    Servo((Servo)):::circular --> RetractAirbrakes
+    Servo --> ExtendAirbrakes
```
Comment (harshil21) [inline README.md:132]: can you change this so that the arrow heads go to Servo instead of from the Servo?

File: README.md [hunk 4]
```diff
@@ -57,10 +57,87 @@ AirbrakesV2/
...
+    Servo --> Airbrakes
+
+```
+| Type    | Color    | Examples                                                                                                         |
+|---------|----------|------------------------------------------------------------------------------------------------------------------|
+| Classes | ![#44007e](https://via.placeholder.com/15/44007e/000000?text=+) `#44007e`  | main.py, Airbrakes Context, State, Logger, IMU, IMU Data Processor, Servo, Flight, Standby, Freefall, Landed, Motor Burn |
```
Comment (harshil21) [inline README.md:138]:
```suggestion
| Classes | ![#44007e](https://via.placeholder.com/15/44007e/000000?text=+) `#44007e`  | Airbrakes Context, State, Logger, IMU, IMU Data Processor, Servo, Flight, Standby, Freefall, Landed, Motor Burn |
```
technically main.py isn't a class. Maybe you can add another color for it?

File: README.md [hunk 5]
```diff
@@ -57,10 +57,87 @@ AirbrakesV2/
...
+    %% IMU Data Processing
+    IMUDataPacket --> IMUDataProcessor[IMU Data Processor]:::circular
+    IMUDataProcessor --> ProcessedData[(Processed Data Packet)]:::outputSquare
+    IMUDataProcessor --> Speed[(Speed)]:::outputSquare
+    IMUDataProcessor --> Attitude[(Attitude)]:::outputSquare
+    IMUDataProcessor --> AvgAccel[(Avg Acceleration)]:::outputSquare
+    
+    Speed --> ProcessedData
+    Attitude --> ProcessedData
+    AvgAccel --> ProcessedData
```
Comment (harshil21) [inline README.md:117]:
```suggestion
    IMUDataProcessor --> Altitude[(Altitude)]:::outputSquare
    IMUDataProcessor --> AvgAccel[(Avg Acceleration)]:::outputSquare
    
    Speed --> ProcessedData
    Altitude --> ProcessedData
    AvgAccel --> ProcessedData
```

File: README.md [hunk 6]
```diff
@@ -57,10 +57,87 @@ AirbrakesV2/
...
+    MotorBurnState((Motor Burn)):::circular --> State
+    
+    %% Connections with Labels
+    Airbrakes ---|Child Process| Logger((Logger)):::circular
+    Airbrakes ---|Child Process| IMU((IMU)):::circular
+    IMU((IMU)):::circular ---|Fetch Packets| IMUDataPacket[(IMU Data Packet)]:::outputSquare
+
+    %% IMU Data Processing
+    IMUDataPacket --> IMUDataProcessor[IMU Data Processor]:::circular
+    IMUDataProcessor --> ProcessedData[(Processed Data Packet)]:::outputSquare
```
Comment (harshil21) [inline README.md:110]:
```suggestion
```

let's remove this connection

File: README.md (hunk 1 above)
Comment (Danuragtiwari) [reply README.md]: removed it in the latest change!

**Comment by Danuragtiwari** [comment]
@harshil21 I have made changes that you were mentioned previously, please review it and let me know if anything needed!

File: README.md [hunk 7]
```diff
@@ -60,7 +131,7 @@ AirbrakesV2/
 ├── pyproject.toml [configuration file for the project]
 ├── README.md
 ```
-
+                                                  |
```
Comment (harshil21) [inline README.md:134]:
```suggestion
```

**Review by harshil21 — COMMENTED** [review]
Looks good!

**Review by harshil21 — APPROVED** [review]
Thanks for the contribution!
